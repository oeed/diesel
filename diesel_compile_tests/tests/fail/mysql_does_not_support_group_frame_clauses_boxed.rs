extern crate diesel;

use diesel::*;

table! {
    users {
        id -> Integer,
        name -> VarChar,
    }
}

fn main() {
    let mut connection = MysqlConnection::establish("").unwrap();

    let res = users::table
        .select(Box::new(
            dsl::count(users::id).frame_by(
                dsl::frame::Rows
                    .frame_start_with_exclusion(dsl::frame::CurrentRow, dsl::frame::ExcludeGroup),
            ),
        ) as Box<dyn BoxableExpression<_, _, SqlType = _>>)
        //~^^^^^^ ERROR: `ExcludeGroup` is no valid SQL fragment for the `Mysql` backend
        .load::<i64>(&mut connection)
        .unwrap();
}
