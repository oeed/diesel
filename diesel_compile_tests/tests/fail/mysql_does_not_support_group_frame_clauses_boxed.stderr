error[E0277]: `ExcludeGroup` is no valid SQL fragment for the `Mysql` backend
  --> tests/fail/mysql_does_not_support_group_frame_clauses_boxed.rs:16:17
   |
LL |           .select(Box::new(
   |  _________________^
LL | |             dsl::count(users::id).frame_by(
LL | |                 dsl::frame::Rows
LL | |                     .frame_start_with_exclusion(dsl::frame::CurrentRow, dsl::frame::ExcludeGroup),
LL | |             ),
LL | |         ) as Box<dyn BoxableExpression<_, _, SqlType = _>>)
   | |_________^ the trait `QueryFragment<Mysql, NoFrameFrameExclusionSupport>` is not implemented for `ExcludeGroup`
   |
   = note: this usually means that the `Mysql` database system does not support 
           this SQL syntax
   = help: the following other types implement trait `QueryFragment<DB, SP>`:
             `ExcludeGroup` implements `QueryFragment<DB, FrameExclusionSupport>`
             `ExcludeGroup` implements `QueryFragment<DB>`
   = note: required for `ExcludeGroup` to implement `QueryFragment<Mysql>`
   = note: 4 redundant requirements hidden
   = note: required for `AggregateExpression<count<Integer, id>, NoPrefix, NoOrder, ..., ...>` to implement `QueryFragment<Mysql>`
   = note: required for `AggregateExpression<count<Integer, id>, NoPrefix, NoOrder, ..., ...>` to implement `diesel::BoxableExpression<users::table, Mysql>`
   = note: required for the cast from `Box<AggregateExpression<count<Integer, id>, NoPrefix, ..., ..., ...>>` to `Box<dyn diesel::BoxableExpression<users::table, Mysql, SqlType = BigInt>>`

   For more information about this error, try `rustc --explain E0277`.
